server/lib/util.js

```diff
@@ -118,3 +119,19 @@ function dateFormatter(rows, dateFormat) {
 
   return rows;
 }
+
+// this function separate amounts by curreny_id
+
+function sumAmounts(rows) {
+  let tb = {};
+  for (let i = 0; i < rows.length; i++) {
+    let row = rows[i];
+    if (tb[row.currency_id]) {
+      tb[row.currency_id].amount += row.amount;
+      tb[row.currency_id].iteration += 1;
```

Comment:
Could rename `iteration` by something which more meaningful in this context such as `count`